backend-project/small_eod/institutions/tests/test_serializers.py

```diff
@@ -0,0 +1,174 @@
+
+    def test_field_validation_inheritance(self):
+        serializer = ExternalIdentifierNestedSerializer(
+            data={"nip": "12345678901", "regon": "1234567890"}
+        )
+        self.assertFalse(serializer.is_valid(), serializer.errors)
+
+    def test_update(self):
+        serializer = ExternalIdentifierNestedSerializer(
+            self.obj, data={"nip": "1111111111"}
+        )
+        self.assertTrue(serializer.is_valid(), serializer.errors)
+        obj = serializer.save()
+        self.assertEqual(serializer.data["nip"], "1111111111")
+        self.assertEqual(obj.nip, "1111111111")
+
+
+class InstitutionSerializerTestCase(TestCase):
+    def setUp(self):
+        self.user = UserFactory()
+        factory = APIRequestFactory()
+        self.request = factory.get("/")
+        force_authenticate(self.request, user=self.user)
+        self.request.user = self.user
+        self.category = CategoryFactory(level=3)
+        self.admin_unit = JednostkaAdministracyjnaFactory(category=self.category)
+        self.default_serializer = InstitutionSerializer(
+            data=self.get_default_data(), context={"request": self.request}
+        )
+        self.obj = (
+            self.default_serializer.save()
+            if self.default_serializer.is_valid()
+            else None
+        )
+
+    def get_default_data(self, new_data=None, skip=None):
+        new_data = new_data or {}
+        skip = skip or []
+        default_data = {
+            "name": "Polska Fundacja Narodowa o rejestr umów",
+            "administrative_unit": self.admin_unit.pk,
+            "address": {
+                "email": "test@test.test",
+                "city": "Dzierżoniów",
+                "epuap": "asdfg",
+                "house_no": "666",
+            },
+            "external_identifier": {"nip": "1234567890", "regon": "1234567890"},
+        }
+        for field in skip:
+            del default_data[field]
+        return {
+            **default_data,
+            **new_data,
+        }
+
+    def test_nested_address_fields(self):
+        self.assertEqual(self.obj.address.city, "Dzierżoniów")
+        data = InstitutionSerializer(Institution.objects.get()).data
+        self.assertTrue(data["address"]["house_no"], "666")
+
+    def test_nested_external_identifier_fields(self):
+        self.assertEqual(self.obj.external_identifier.nip, "1234567890")
+        data = InstitutionSerializer(Institution.objects.get()).data
+        self.assertTrue(data["external_identifier"]["regon"], "1234567890")
+
+    def test_invalid_administrative_unit(self):
+        admin_unit = JednostkaAdministracyjnaFactory()
+        serializer = InstitutionSerializer(
+            data=self.get_default_data(new_data={"administrative_unit": admin_unit.pk}),
+            context={"request": self.request},
+        )
+        self.assertFalse(serializer.is_valid(), serializer.errors)
+
+    def test_update(self):
```

Comment:
```suggestion
    def test_update_name(self):
```

Detal, ale będzie prowokować, aby kolejne pola jako osobne przypadki dopisywać.

Reply:
Zrobione